migrate.go

```diff
@@ -25,8 +25,30 @@ const (
 	Down
 )
 
-var tableName = "gorp_migrations"
-var schemaName = ""
+// MigrationSet provides database parameters for a migration execution
+type MigrationSet struct {
+	// Name of the table used to store migration info.
+	tableName string
+	// Name of a schema that the migration table be referenced.
+	schemaName string
+}
+
+// NewMigrationSet returns a parametrized Migration object
+func NewMigrationSet(schemaName string, tableName string) MigrationSet {
+	if tableName == "" {
+		tableName = "gorp_migrations"
+	}
+	return MigrationSet{
+		tableName:  tableName,
+		schemaName: schemaName,
+	}
+}
```

Comment:
My only concern here is that an empty `MigrationSet` is an invalid configuration.

Usually in go the empty value of a struct is a useful one. Perhaps it's better to remove the `tableName == ""` check here and move it to where the value is actually used.

Once you do that, the members of `MigrationSet` can become public, which will allow users to use a custom one where they only override the schema name:

```
MigrationSet{
    SchemaName: "myschema",
}
```

And once we do that, perhaps we don't need `NewMigrationSet()` anymore? Just use the object initializer.

Reply:
To avoid multiple `tableName == ""`, you could define a method `MigrationSet` to "retrieve the `tableName` if set, else default".

Reply:
Oh and this avoids an important risk in terms of backwards compatibility: suppose we ever want to add a third parameter to `MigrationSet`. As is currently, with `NewMigrationSet(string, string)`, we'd need to add a third parameter to that function, which would break any code using it. That's not acceptable. Exposing the struct fields avoids that issue.

Reply:
Seems legit to me, I'll propose a commit in this direction